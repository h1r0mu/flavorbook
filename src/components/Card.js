import React from "react";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
import Avatar from "./Avatar";
import Chip from "./Chips";
import { storage } from "../firebase";
import PropTypes from 'prop-types';

const useStyles = makeStyles(() => ({
  root: {
    maxWidth: 1000,
    display: "flex",
    margin: "auto",
    marginTop: "10px",
    backgroundColor: "white",
    borderRadius: 40,
  },
  details: {
    display: "flex",
    flexDirection: "column",
  },
  content: {
    flex: "1 0 auto",
  },
  cover: {
    margin: 20,
    width: 200,
    height: 200,
    borderRadius: 20,
  },
  coffeeName: {
    display: "flex",
    flexWrap: "no-wrap",
    marginTop: 20,
    marginBottom: 20,
  },
  bold: {
    fontWeight: 600,
  },
  avatar: {
    margin: 0,
  },
  chips: {
    display: "flex",
    flexWrap: "no-wrap",
    marginTop: 30,
  },
  time: {
    marginTop: 20,
    marginLeft: 200,
  },
}));

export default function MediaControlCard(props) {
  const classes = useStyles();
  var imgSample = storage.child("member/" + props.picName);
  imgSample.getDownloadURL().then((downloadURL) => {
    document.getElementById(props.picName).src = downloadURL;
  });
  const flavorItems = props.flavors;
  const chips = flavorItems.map((flavor) => <Chip name={flavor} />);

  return (
    <Card className={classes.root} key={props.coffeeName + props.picName>
      <img
        id={props.picName}
        src=""
        alt={props.picName}
        className={classes.cover}
      />
      <div className={classes.details}>
        <CardContent className={classes.content}>
          <Typography variant="subtitle1" color="textSecondary">
            {props.storeName}
            <ShoppingCartIcon />
          </Typography>
          <div className={classes.coffeeName}>
            <Avatar className={classes.avatar} />
            <Typography component="h3" variant="h5" className={classes.bold}>
              {props.coffeeName}
            </Typography>
          </div>
          <Typography variant="subtitle1" color="textSecondary">
            {props.subtitle}
          </Typography>
          <div className={classes.chips}>
            {chips}
            <Typography
              variant="subtitle1"
              color="textSecondary"
              className={classes.time}
            >
              2021-3-14
            </Typography>
          </div>
        </CardContent>
      </div>
    </Card>
  );
}

Card.propTypes = {
  picName: Proptypes.string,
  flavors: Proptypes.array,
  storeName: Proptypes.string,
  coffeeeName: Proptypes.string,
  subtitle: Proptypes.string,
};
