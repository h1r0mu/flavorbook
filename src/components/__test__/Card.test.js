import React from "react";
import renderer from "react-test-renderer";

import Card from "../Card.js";

it("renders correctly", () => {
  const card = renderer
    .create(
      <Card
        picName="sample.png"
        storeName="はぜや珈琲"
        coffeeName="エチオピア　ハロディ　ナチュラル"
        subtitle="City Roast"
        flavors={["dummy", "dummy"]}
      />
    )
    .toJSON();
  expect(card).toMatchSnapshot();
});
