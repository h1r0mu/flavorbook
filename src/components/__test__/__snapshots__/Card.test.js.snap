// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <img
    alt="sample.png"
    className="makeStyles-cover-4"
    id="sample.png"
    src=""
  />
  <div
    className="makeStyles-details-2"
  >
    <div
      className="MuiCardContent-root makeStyles-content-3"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      >
        はぜや珈琲
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </h6>
      <div
        className="makeStyles-coffeeName-5"
      >
        <div
          className="makeStyles-root-10"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle makeStyles-small-11"
          >
            <img
              alt="Remy Sharp"
              className="MuiAvatar-img"
              src="etiopia.png"
            />
          </div>
        </div>
        <h3
          className="MuiTypography-root makeStyles-bold-6 MuiTypography-h5"
        >
          エチオピア　ハロディ　ナチュラル
        </h3>
      </div>
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      >
        City Roast
      </h6>
      <div
        className="makeStyles-chips-8"
      >
        <div
          className="makeStyles-root-13"
        >
          <div
            className="MuiButtonBase-root MuiChip-root MuiChip-clickable"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <p
              className="MuiChip-icon"
            >
              no icon
            </p>
            <span
              className="MuiChip-label"
            >
              dummy
            </span>
          </div>
        </div>
        <div
          className="makeStyles-root-13"
        >
          <div
            className="MuiButtonBase-root MuiChip-root MuiChip-clickable"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <p
              className="MuiChip-icon"
            >
              no icon
            </p>
            <span
              className="MuiChip-label"
            >
              dummy
            </span>
          </div>
        </div>
        <h6
          className="MuiTypography-root makeStyles-time-9 MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          2021-3-14
        </h6>
      </div>
    </div>
  </div>
</div>
`;
